src/Propel/Runtime/ActiveQuery/QueryExecutor/AbstractQueryExecutor.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+/**
+ * MIT License. This file is part of the Propel package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Propel\Runtime\ActiveQuery\QueryExecutor;
+
+use Propel\Runtime\ActiveQuery\Criteria;
+use Propel\Runtime\ActiveQuery\SqlBuilder\PreparedStatementDto;
+use Propel\Runtime\Connection\ConnectionInterface;
+use Propel\Runtime\Connection\ConnectionWrapper;
+use Propel\Runtime\Connection\StatementWrapper;
+use Propel\Runtime\Propel;
+use Propel\Runtime\ServiceContainer\ServiceContainerInterface;
+use Throwable;
+
+abstract class AbstractQueryExecutor
```

Comment:
This may be a good opportunity to add comments about 'what is the structure of the subsystem these classes represent', i.e.
```
`QueryExecutor`'s responsibility is to translate a `Criteria` object to the specific RDBMS dialect and execute the query.
Use it like this: <insert example here>
```